### ISSUE TITLE
Unable to have more than one profile in fargate_profiles

### ISSUE THREAD

**Issue by brunoyb** [terraform-aws-modules/terraform-aws-eks#1245]
# I have issues

I'm using Terraform v0.14.6. When I try to add a new item to `fargate_profiles` I get an error during `terraform plan`.

## I'm submitting a...

* [X] bug report
* [ ] feature request
* [ ] support request - read the [FAQ](https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/docs/faq.md) first!
* [ ] kudos, thank you, warm fuzzy

## What is the current behavior?

`terraform plan` shows this:

```
Error: Inconsistent conditional result types

  on .terraform/modules/my_eks/modules/fargate/fargate.tf line 17, in resource "aws_eks_fargate_profile" "this":
  17:   for_each               = local.create_eks ? local.fargate_profiles_expanded : {}
    |----------------
    | local.create_eks is true
    | local.fargate_profiles_expanded is object with 4 attributes

The true and false result expressions must have consistent types. The given
expressions are object and object, respectively.
```

## If this is a bug, how to reproduce? Please include a code sample if relevant.

Have an instance of this module with `fargate_profiles`, after a successful `terraform apply` try to add a new item to `fargate_profiles` and run `terraform plan`.

EDIT: Looks like this happens whenever you have more than one fargate profile, not only when adding a new one (see comments below).

## What's the expected behavior?

`terraform plan` should work and show the new `aws_eks_fargate_profile` resources being created.

## Are you able to fix this problem and submit a PR? Link here if you have already.

I don't have a PR because I'm not sure what is correct solution for this, but I made a copy of this module locally and changed this:

```
resource "aws_eks_fargate_profile" "this" {
  for_each               = local.create_eks ? local.fargate_profiles_expanded : {}
```

With this:

```
resource "aws_eks_fargate_profile" "this" {
  for_each               = local.create_eks ? local.fargate_profiles_expanded : null
```

To workaround the issue for now.

In other words, I changed `{}` with `null`. Not even sure how `for_each` works with `null`, but since it didn't complain, I decided to leave it like this until we find the correct fix.

## Environment details

* Affected module version: 14.0.0
* OS: Some recent version of Amazon Linux I believe
* Terraform version: 0.14.6

Thanks!

**Comment by JohnHedman** [comment]
> Are you able to fix this problem and submit a PR? Link here if you have already.
> I don't have a PR because I'm not sure what is correct solution for this, but I made a copy of this module locally and changed this:
> 
> resource "aws_eks_fargate_profile" "this" {
>   for_each               = local.create_eks ? local.fargate_profiles_expanded : {}
> With this:
> 
> resource "aws_eks_fargate_profile" "this" {
>   for_each               = local.create_eks ? local.fargate_profiles_expanded : null
> To workaround the issue for now.
> 
> In other words, I changed {} with null. Not even sure how for_each works with null, but since it didn't complain, I decided to leave it like this until we find the correct fix.

Are there any negative consequences to using `null` instead of `{}`? I am running into this exact same problem with this module and I have to deploy multiple Fargate profiles for my use case. Seems like as soon as you introduce more than one Fargate profile, this module will run into this error.

Module version: 17.0.3
Terraform version: 0.15.4
OS: Amazon Linux 2

**Comment by signaleleven** [comment]
I haven't tried the suggested solution yet (to be honest, I'd rather not cloning the module in the project I need it for), but I confirm that this happens not only when you try to add a profile to an existing cluster, but even attempting to create a cluster with more than one fargate profile.
Therefore the issue could be renamed to reflect the fact that - currently - the module does not support more than one fargate profile.

**Comment by JohnHedman** [comment]
> I haven't tried the suggested solution yet (to be honest, I'd rather not cloning the module in the project I need it for), but I confirm that this happens not only when you try to add a profile to an existing cluster, but even attempting to create a cluster with more than one fargate profile.
> Therefore the issue could be renamed to reflect the fact that - currently - the module does not support more than one fargate profile.

Yeah, I can confirm this as well. The module runs into this error when more than one fargate profile is specified.

**Comment by antonbabenko** [comment]
[v17.19.0](https://github.com/terraform-aws-modules/terraform-aws-eks/releases/v17.19.0) has been just released with this fix. See `examples/fargate` for more information.
